### PR TITLE
change of the project to NextJS

### PR DESCRIPTION
**Pull request by Ksenaaa**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1 +1 @@
-REACT_APP_API_URL='http://localhost:3005'
+NEXT_PUBLIC_API_URL='http://localhost:3005'
```
Comment (vell28) [inline .env:1]: PUBLIC_API_URL

File: next.config.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const { i18n } = require('./next-i18next.config')
+
+/** @type {import('next').NextConfig} */
```
Comment (vell28) [inline next.config.js:3]: do we really need /** @type {import('next').NextConfig} */ ?

File: src/components/Header/Header.tsx [hunk 3]
```diff
@@ -8,25 +10,21 @@ import { MiddleHeader } from './MiddleHeader'
 import styles from './Header.module.scss'
 
 export const Header = () => {
-    const [menuBurgerActive, setMenuBurgerActive] = useState(false)
-
-    const onClickBurger = () => {
-        setMenuBurgerActive(preState => !preState)
-    }
+  const { isOpen: menuBurgerActive, onToggle: setMenuBurgerActive } = useToggle()
```
Comment (vell28) [inline src/components/Header/Header.tsx]: setMenuBurgerActive  - again? does `setMenuBurgerActive ` set only Active ?

Comment (vell28) [reply src/components/Header/Header.tsx]: be more attentive please! You do the same mistakes.

File: src/components/Header/Menu/Menu.module.scss [hunk 4]
```diff
@@ -1,68 +1,89 @@
...
-
-        &:first-letter {
-            text-transform: uppercase;
-        }
+  margin: 0 auto;
+  display: flex;
+  justify-content: space-between;
+  
+  .menuItem {
+    & a {
```
Comment (vell28) [inline src/components/Header/Menu/Menu.module.scss]: & a = a

File: src/components/Header/Menu/Menu.tsx [hunk 5]
```diff
@@ -1,38 +1,49 @@
...
+  const { t, i18n } = useTranslation()
+
+  const router = useRouter()
+
+  const menuName = t('menu', { returnObjects: true }) as menuNameType[]
+
+  return (
+    <div className={cn(styles.wrapper, !isActiveMenuBurger ? styles.wrapperMenu : styles.wrapperActiveMenuBurger)}>
+      <ul className={styles.menuList}>
+        {Array.isArray(menuName) && menuName.map(item =>
```
Comment (vell28) [inline src/components/Header/Menu/Menu.tsx]: what does it mean? or set truly  type fro `menuName` or write normal condition

File: src/components/Header/Menu/Menu.tsx [hunk 6]
```diff
@@ -1,38 +1,49 @@
...
+import { useTranslation } from 'next-i18next'
 
 import { LanguageName } from 'utils/constants/languages'
 
 import styles from './Menu.module.scss'
 
 type Props = {
-    isActiveMenu?: boolean;
+  isActiveMenuBurger?: boolean;
+  onClickItemMenu?: () => void;
```
Comment (vell28) [inline src/components/Header/Menu/Menu.tsx]: onClick

File: src/components/Header/Menu/Menu.tsx [hunk 7]
```diff
@@ -1,38 +1,49 @@
...
+
+  const menuName = t('menu', { returnObjects: true }) as menuNameType[]
+
+  return (
+    <div className={cn(styles.wrapper, !isActiveMenuBurger ? styles.wrapperMenu : styles.wrapperActiveMenuBurger)}>
+      <ul className={styles.menuList}>
+        {Array.isArray(menuName) && menuName.map(item =>
+          <li
+            key={item.id}
+            onClick={onClickItemMenu}
```
Comment (vell28) [inline src/components/Header/Menu/Menu.tsx]: onClick

File: src/components/Header/Menu/Menu.tsx [hunk 8]
```diff
@@ -1,38 +1,49 @@
...
+            className={cn(styles.menuItem, {
+              [styles.menuItemUa]: i18n.language === LanguageName.UA,
+              [styles.activeMenuItem]: router.pathname.split('/').includes(item.path)
+            })}
+          >
+            <Link href={`/catalog/${item.path}`}>{item.name}</Link>
```
Comment (vell28) [inline src/components/Header/Menu/Menu.tsx]: use link from routing!!

File: src/components/Header/MiddleHeader/MiddleHeader.tsx [hunk 9]
```diff
@@ -14,33 +17,33 @@ import { Search } from '../Search'
...
-                <div className={styles.heart}>
-                    <img src={heartIcon} alt="heart" className={styles.blueHeart}/>
-                    <img src={orangeHeartIcon} alt="heart" className={styles.orangeHeart}/>
-                </div>
-                <div className={styles.basket}>
-                    <img src={basketIcon} alt="basket" className={styles.blueBasket}/>
-                    <img src={orangeBasketIcon} alt="basket" className={styles.orangeBasket}/>
-                </div>
-            </div>
+  const [activeSearch, setActiveSearch] = useState(false)
```
Comment (vell28) [inline src/components/Header/MiddleHeader/MiddleHeader.tsx]: toggle use

Comment (vell28) [reply src/components/Header/MiddleHeader/MiddleHeader.tsx]: pls think before you do something!!

File: src/components/Loader/Loader.module.scss [hunk 10]
```diff
@@ -1,61 +1,61 @@
...
+        animation: ldsEllipsis1 0.6s infinite;
+      }
 
-            &:nth-child(2) {
-                left: 8px;
-                animation: ldsEllipsis2 0.6s infinite;
-            }
+      &:nth-child(2) {
+        left: 8px;
+        animation: ldsEllipsis2 0.6s infinite;
```
Comment (vell28) [inline src/components/Loader/Loader.module.scss]: use const for animation time

File: src/components/Loader/Loader.module.scss [hunk 11]
```diff
@@ -1,61 +1,61 @@
...
+      }
 
-            &:nth-child(3) {
-                left: 32px;
-                animation: ldsEllipsis2 0.6s infinite;
-            }
+      &:nth-child(3) {
+        left: 32px;
+        animation: ldsEllipsis2 0.6s infinite;
```
Comment (vell28) [inline src/components/Loader/Loader.module.scss]: the same, and check averywhere

File: src/components/ModalWindow/ModalWindow.tsx [hunk 12]
```diff
@@ -7,33 +7,33 @@ import { ModalWindowItem } from './ModalWindowItem'
 import styles from './ModalWindow.module.scss'
 
 type Props = {
-    modalList: string[],
-    isActive: string,
-    onChangeItem: (item: string) => void,
-    onToggleModal: () => void,
-    icon: boolean
+  modalList: string[],
+  isActive: string,
```
Comment (vell28) [inline src/components/ModalWindow/ModalWindow.tsx]: really? isActive: string

File: src/components/ModalWindow/ModalWindow.tsx [hunk 13]
```diff
@@ -7,33 +7,33 @@ import { ModalWindowItem } from './ModalWindowItem'
...
+  onToggleModal: () => void,
+  icon: boolean
 }
 
 export const ModalWindow: FC<Props> = ({ modalList, isActive, onChangeItem, onToggleModal, icon }) => {
-    const onClickItem = useCallback((itemModal: string) => {
-        onChangeItem(itemModal)
-        onToggleModal()
-    }, [])
+  const onClickItem = useCallback((itemModal: string) => {
```
Comment (vell28) [inline src/components/ModalWindow/ModalWindow.tsx]: maybe itemName?

File: src/components/ModalWindow/ModalWindow.tsx [hunk 14]
```diff
@@ -7,33 +7,33 @@ import { ModalWindowItem } from './ModalWindowItem'
...
-                )}
-            </ul>
-            <div className={styles.substrate} onClick={onToggleModal}></div>
-        </>
-    )
+  return (
+    <>
+      <ul className={cn(styles.modalList, icon && styles.modalListWithIcon)}>
+        {modalList.map((item, i) =>
+          <ModalWindowItem
```
Comment (vell28) [inline src/components/ModalWindow/ModalWindow.tsx]: What kind of item? set more clear name to component

File: src/components/ModalWindow/ModalWindowItem.tsx [hunk 15]
```diff
@@ -8,22 +10,22 @@ import arrow from 'img/svg/arrow-black.svg'
...
 
 type Props = {
-    item: string,
-    isActive: string,
-    onClickItem: (item: string) => void,
-    icon: boolean
+  item: string,
+  isActive: string,
+  onClickItem: (item: string) => void,
+  icon: boolean
```
Comment (vell28) [inline src/components/ModalWindow/ModalWindowItem.tsx]: is it correct?

File: src/components/ModalWindow/ModalWindowItem.tsx [hunk 16]
```diff
@@ -8,22 +10,22 @@ import arrow from 'img/svg/arrow-black.svg'
...
 }
 
 export const ModalWindowItem: FC<Props> = ({ item, isActive, onClickItem, icon }) => {
-    const onClick = () => onClickItem(item)
+  const onClick = () => onClickItem(item)
```
Comment (vell28) [inline src/components/ModalWindow/ModalWindowItem.tsx]: when we create event hendler function inside component we set name `handle` or `handler`! if pass as props we set `on`

File: src/components/ModalWindow/ModalWindowItem.tsx [hunk 17]
```diff
@@ -8,22 +10,22 @@ import arrow from 'img/svg/arrow-black.svg'
...
-            className={cn(styles.modalItem, { [styles.isActive]: isActive === item })}
-            onClick={onClick}
-        >
-            {item}
-            {icon && <img src={isActive !== item ? arrow : arrowActive} alt="arrow" />}
-        </li>
-    )
+  return (
+    <li
+      className={cn(styles.modalItem, { [styles.isActive]: isActive === item })}
```
Comment (vell28) [inline src/components/ModalWindow/ModalWindowItem.tsx]: isActive not correct name!

File: src/pages/Home.module.scss [hunk 18]
```diff
@@ -0,0 +1,2 @@
+.wrapper {
```
Comment (vell28) [inline src/pages/Home.module.scss]: ?

**Review by vell28 — APPROVED** [review]

File: .env (hunk 1 above)
Comment (Ksenaaa) [reply .env:1]: discussed

File: next.config.js (hunk 2 above)
Comment (Ksenaaa) [reply next.config.js:3]: discussed

File: src/components/Header/Header.tsx (hunk 3 above)
Comment (Ksenaaa) [reply src/components/Header/Header.tsx]: fixed

File: src/components/Header/Menu/Menu.module.scss (hunk 4 above)
Comment (Ksenaaa) [reply src/components/Header/Menu/Menu.module.scss]: fixed

File: src/components/Header/Menu/Menu.tsx (hunk 5 above)
Comment (Ksenaaa) [reply src/components/Header/Menu/Menu.tsx]: fixed

File: src/components/Header/Menu/Menu.tsx (hunk 6 above)
Comment (Ksenaaa) [reply src/components/Header/Menu/Menu.tsx]: fixed

File: src/components/Header/Menu/Menu.tsx (hunk 7 above)
Comment (Ksenaaa) [reply src/components/Header/Menu/Menu.tsx]: fixed

File: src/pages/Home.module.scss (hunk 18 above)
Comment (Ksenaaa) [reply src/pages/Home.module.scss]: fixed

File: src/components/Loader/Loader.module.scss (hunk 11 above)
Comment (Ksenaaa) [reply src/components/Loader/Loader.module.scss]: fixed

File: src/components/Header/MiddleHeader/MiddleHeader.tsx (hunk 9 above)
Comment (Ksenaaa) [reply src/components/Header/MiddleHeader/MiddleHeader.tsx]: fixed

File: src/components/Loader/Loader.module.scss (hunk 10 above)
Comment (Ksenaaa) [reply src/components/Loader/Loader.module.scss]: fixed

File: src/components/ModalWindow/ModalWindow.tsx (hunk 13 above)
Comment (Ksenaaa) [reply src/components/ModalWindow/ModalWindow.tsx]: fixed

File: src/components/ModalWindow/ModalWindow.tsx (hunk 14 above)
Comment (Ksenaaa) [reply src/components/ModalWindow/ModalWindow.tsx]: fixed

File: src/components/ModalWindow/ModalWindowItem.tsx (hunk 15 above)
Comment (Ksenaaa) [reply src/components/ModalWindow/ModalWindowItem.tsx]: fixed

File: src/components/ModalWindow/ModalWindowItem.tsx (hunk 17 above)
Comment (Ksenaaa) [reply src/components/ModalWindow/ModalWindowItem.tsx]: fixed

File: src/components/ModalWindow/ModalWindow.tsx (hunk 12 above)
Comment (Ksenaaa) [reply src/components/ModalWindow/ModalWindow.tsx]: fixed

File: src/components/ModalWindow/ModalWindowItem.tsx (hunk 16 above)
Comment (Ksenaaa) [reply src/components/ModalWindow/ModalWindowItem.tsx]: fixed

File: src/components/Header/Menu/Menu.tsx (hunk 8 above)
Comment (Ksenaaa) [reply src/components/Header/Menu/Menu.tsx]: fixed

File: src/components/Header/Menu/constants/pathToParent.ts [hunk 19]
```diff
@@ -0,0 +1,3 @@
+export enum pathToParent {
```
Comment (vell28) [inline src/components/Header/Menu/constants/pathToParent.ts]: you need to create variable with all route paths

File: src/components/Modal/Modal.tsx [hunk 20]
```diff
@@ -0,0 +1,39 @@
...
+import { ModalItem } from './ModalItem'
+
+import styles from './Modal.module.scss'
+
+type Props = {
+  modalList: string[],
+  activeItem: string,
+  onChangeItem: (item: string) => void,
+  onToggleModal: () => void,
+  isThereIcon: boolean
```
Comment (vell28) [inline src/components/Modal/Modal.tsx]: isIconVisible

File: src/styles/_animation.scss [hunk 21]
```diff
@@ -1,2 +1,3 @@
 $animationTransitionTime: .3s all;
```
Comment (vell28) [inline src/styles/_animation.scss]: rename

**Review by vell28 — APPROVED** [review]

File: src/components/Header/Menu/constants/pathToParent.ts (hunk 19 above)
Comment (Ksenaaa) [reply src/components/Header/Menu/constants/pathToParent.ts]: fixed

File: src/components/Modal/Modal.tsx (hunk 20 above)
Comment (Ksenaaa) [reply src/components/Modal/Modal.tsx]: fixed

File: src/styles/_animation.scss (hunk 21 above)
Comment (Ksenaaa) [reply src/styles/_animation.scss]: fixed

**Review by vell28 — APPROVED** [review]

**Pull request merged**